llvm/lib/Target/DirectX/DXILShaderFlags.cpp

```diff
@@ -206,14 +213,23 @@ void ModuleShaderFlags::initialize(Module &M, DXILResourceTypeMap &DRTM,
                                    const ModuleMetadataInfo &MMDI) {
 
   CanSetResMayNotAlias = MMDI.DXILVersion >= VersionTuple(1, 7);
-
-  // Check if -res-may-alias was provided on the command line.
-  // The command line option will set the dx.resmayalias module flag to 1.
+  // The command line option -res-may-alias will set the dx.resmayalias module
+  // flag to 1 and disable the ability to set the ResMayNotAlias flag
   if (auto *RMA = mdconst::extract_or_null<ConstantInt>(
           M.getModuleFlag("dx.resmayalias")))
     if (RMA->getValue() != 0)
       CanSetResMayNotAlias = false;
 
+  // NativeLowPrecisionMode can only be set when the command line option
+  // -enable-16bit-types is provided. This is indicated by the dx.nativelowprec
+  // module flag being set
+  NativeLowPrecisionMode = false;
+  if (auto *NativeLowPrec = mdconst::extract_or_null<ConstantInt>(
+          M.getModuleFlag("dx.nativelowprec")))
+    if (MMDI.ShaderModelVersion >= VersionTuple(6, 2) &&
+        NativeLowPrec->getValue() != 0)
+      NativeLowPrecisionMode = true;
```

Comment:
I think it's slightly clearer to assign the boolean value directly here:
```suggestion
    if (MMDI.ShaderModelVersion >= VersionTuple(6, 2))
      NativeLowPrecisionMode = NativeLowPrec->getValue().getBoolValue();
```